rali/rali/source/meta_node_crop_resize.cpp

```diff
@@ -0,0 +1,106 @@
+/*
+Copyright (c) 2019 - 2020 Advanced Micro Devices, Inc. All rights reserved.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#include "meta_node_crop_resize.h"
+void CropResizeMetaNode::initialize()
+{
+    _x1_val.resize(_batch_size);
+    _y1_val.resize(_batch_size);
+    _x2_val.resize(_batch_size);
+    _y2_val.resize(_batch_size);
+}
+
+void CropResizeMetaNode::update_parameters(MetaDataBatch* input_meta_data)
+{
+    initialize();
+    if(_batch_size != input_meta_data->size())
+    {
+        _batch_size = input_meta_data->size();
+    }
+    _meta_crop_param = _node->get_crop_param();    
+    _dst_width = _node->get_dst_width();
+    _dst_height = _node->get_dst_height();
+    _x1 = _meta_crop_param->x1_arr;
+    _y1 = _meta_crop_param->y1_arr;
+    _x2 = _meta_crop_param->x2_arr;
+    _y2 = _meta_crop_param->y2_arr;
+    vxCopyArrayRange((vx_array)_x1, 0, _batch_size, sizeof(uint),_x1_val.data(), VX_READ_ONLY, VX_MEMORY_TYPE_HOST);
+    vxCopyArrayRange((vx_array)_y1, 0, _batch_size, sizeof(uint),_y1_val.data(), VX_READ_ONLY, VX_MEMORY_TYPE_HOST);
+    vxCopyArrayRange((vx_array)_x2, 0, _batch_size, sizeof(uint),_x2_val.data(), VX_READ_ONLY, VX_MEMORY_TYPE_HOST);
+    vxCopyArrayRange((vx_array)_y2, 0, _batch_size, sizeof(uint),_y2_val.data(), VX_READ_ONLY, VX_MEMORY_TYPE_HOST);
+    unsigned bb_count;
```

Comment:
Codacy found an issue: [The scope of the variable 'bb_count' can be reduced.](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/pullRequest?prid=5976200)